src/cargo/ops/cargo_compile.rs

```diff
@@ -887,19 +882,19 @@ fn generate_targets(
             };
 
             let is_local = pkg.package_id().source_id().is_path();
-            let profile = profiles.get_profile(
-                pkg.package_id(),
-                ws.is_member(pkg),
-                is_local,
-                unit_for,
-                target_mode,
-            );
 
             // No need to worry about build-dependencies, roots are never build dependencies.
             let features_for = FeaturesFor::from_for_host(target.proc_macro());
             let features = resolved_features.activated_features(pkg.package_id(), features_for);
 
             for kind in requested_kinds {
+                let profile = profiles.get_profile(
+                    pkg.package_id(),
+                    ws.is_member(pkg),
+                    is_local,
+                    unit_for,
+                    target_mode,
+                );
```

Comment:
It's not clear to me why this moved inside the loop.

Reply:
Oh this was a relic of when this was doing a macOS-specific thing internally so the `CompileKind` was passed in, I undid that though so I just undid this as well.